File: DataAccessLayerWriter/StringExtensions.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataAccessLayerWriter
{
    static class StringExtensions
    {
        public static string RemoveSquareBrackets(this string item)
            => item?.Replace("[", "").Replace("]", "");


        public static string Join(this IEnumerable<string> items, string separator) => String.Join(separator, items);
        public static string Join(this IEnumerable<char> items, string separator) => String.Join(separator, items);

        static string MakeWordPascal(string word) => (String.IsNullOrEmpty(word)) 
                ? word : word[0].ToString().ToUpper() + word.Skip(1).Join("");

        static string MakeWordCamel (string word) => (String.IsNullOrEmpty(word)) 
                ? word : word[0].ToString().ToLower() + word.Skip(1).Join("");

        static string FormatStatement (string item, Func<string,string> converter) => 
            item.Replace("@", "").Split(new[] { '_', '_', '-', ' ' }, StringSplitOptions.None).Select(converter).Join(",");

        public static string ToPascalCase(this string item) => FormatStatement(item, MakeWordPascal);

        public static string ToCamelCase (this string item) => FormatStatement(item, MakeWordCamel);


    }
}

File: DataAccessLayerWriterToFSharp/FSharp/StringExtensions.fs
namespace DataAccessLayerWriter

    open System;


    module StringExtensions = 


        type System.String with 
            member this.RemoveSquareBrackets() =
                if isNull this then 
                    null
                else 
                    this.Replace("[", "").Replace("]", "");

            member this.MakeWordPascal() = 
                if String.IsNullOrEmpty(this) then 
                    this
                else 
                    this.[0].ToString().ToUpper() + (Seq.skip 1 word);


            member this.MakeWordCamel () = (String.IsNullOrEmpty(word)) 
                ? word : word[0].ToString().ToLower() + word.Skip(1).Join("");

            member item.FormatStatement (converter:string -> string) =  
                item.Replace("@", "").Split([| '_', '_', '-', ' ' |], StringSplitOptions.None).
                    Select(converter).Join(",");

            member item.ToPascalCase() = item.FormatStatement(item.MakeWordPascal);

            member item.ToCamelCase () = FormatStatement(item, MakeWordCamel);


        (*

            
        member this.Join(this IEnumerable<string> items, string separator) = 
            String.Join(separator, items);

        let openJoin(this IEnumerable<char> items, string separator) = 
            String.Join(separator, items);

        *)


    
